test/Base_Tests/src/Semantic/Warnings_Spec.enso

```diff
@@ -461,6 +463,15 @@ add_specs suite_builder = suite_builder.group "Dataflow Warnings" group_builder-
             caught_panic.payload.to_text
         panic_text . should_equal "Type error: expected `warnings` to be Vector, but got Integer."
 
+    group_builder.specify "warning & auto constructor" <|
+        three = Warning.attach "warn!!!" 3
+        wrap = Wrap.materialize (..Value three)
```

Comment:
Probably not directly related to this PR, but how does `..Value` autoscoped constructor resolve to `Wrap.Value`? There is also `My_Warning.Value` in the scope, and `materialize` does not provide type signature for its argument. Is it because of the return value?

Reply:
- Yes, the resolution of the auto scoped constructor happens when checking the return value of `materialize`
- this kind of behavior would be "void" if #12292 is implemented
- then we would need to type check the argument.